framework/src/meshmodifiers/MeshExtruder.C

```diff
@@ -64,18 +73,18 @@ MeshExtruder::modify()
 
   _mesh_ptr->getMesh().clear();
 
-  QueryElemSubdomainID elem_subdomain_id(getParam<std::vector<SubdomainID> >("existing_subdomains"),
-                                         getParam<std::vector<unsigned int> >("layers"),
-                                         getParam<std::vector<unsigned int> >("new_ids"),
-                                         _num_layers);
+  QueryElemSubdomainID * elem_subdomain_id = NULL;
+  if (_map_custom_ids)
+    elem_subdomain_id = new QueryElemSubdomainID(_existing_subdomains, _layers, _new_ids, _num_layers);
```

Comment:
Aoid raw pointers with dynamic memory. Use `std::unique_ptr` and `libmesh_make_unique(args)`